src/Tests/UriTemplates/Compilers/Parsers/Lexers/UriTemplateLexerTest.php

```diff
@@ -68,13 +68,24 @@ public function testLexingPathWithDefaultValueAndRules(): void
         );
     }
 
-    public function testLexingPathWithFloat(): void
+    public function lexingPathWithFloatProvider(): array
+    {
+        return [
+            [1.23, '1.23'],
+            [123.0, '123.0'],
+        ];
+    }
+
+    /**
+     * @dataProvider lexingPathWithFloatProvider
+     */
+    public function testLexingPathWithFloat($number, $expectedValue): void
```

Comment:
Hi @opulencephp, thank you for your reply. I've added the two test cases for the float/integer value lexer in `UriTemplateLexerTest` class.